### PR TITLE
GDB-4836 support empty prefix

### PR DESCRIPTION
**Pull request by svilenvelikov**
* Add namespaces type and a namespaces service collecting functions for working with the namespaces
* Apply namespace type and service's functions where needed
* Allow the empty prefix sign to be visible in the UI
* Add tests

Resolves:
https://ontotext.atlassian.net/browse/GDB-4836
https://ontotext.atlassian.net/browse/GDB-4869

### CODE REVIEW

File: src/app/main/mapper/mapper-dialog/mapper-dialog.component.ts [hunk 1]
```diff
@@ -516,11 +526,6 @@ export class MapperDialogComponent extends OnDestroyMixin implements OnInit {
         }));
   }
 
-  private filterNamespace(value: string): object[] {
```
Comment (svilenvelikov) [inline src/app/main/mapper/mapper-dialog/mapper-dialog.component.ts:519]: This was obsolete so I just deleted it

File: src/app/services/model-construct.service.ts [hunk 2]
```diff
@@ -165,24 +168,25 @@ export class ModelConstructService {
...
       return this.getPrefixTransformation(t, namespaces);
     });
   }
 
-  private isAllowedExpression(expression: string, language: string): boolean {
+  private isAllowedExpression(expression: string): boolean {
     // Allow the empty prefix
-    return !Helper.isBlank(expression) || (language === Language.Prefix.valueOf() && expression === '');
+    return !Helper.isBlank(expression) || expression === ':';
   }
```
Comment (teodossidossev) [inline src/app/services/model-construct.service.ts:190]: there is a constant for colon

File: src/app/services/namespace.service.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  static addNamespace(namespaces: Namespaces, namespace: Namespace): void {
+    if (namespace) {
+      namespaces[namespace.prefix] = namespace.value;
+    }
+  }
+
+  static toNamespace(str: string): Namespace | undefined {
+    if ((str.split('=').length > 1)) {
+      const split = str.split(/=(.+)/);
```
Comment (teodossidossev) [inline src/app/services/namespace.service.ts:36]: you can extend this regex to a class const as we'll have to change it shortly

File: src/app/services/model-construct.service.ts (hunk 2 above)
Comment (svilenvelikov) [reply src/app/services/model-construct.service.ts:190]: Thx, I'll use it

File: src/app/services/namespace.service.ts (hunk 3 above)
Comment (svilenvelikov) [reply src/app/services/namespace.service.ts:36]: If I make it a constant I'll be forced to use the RegExp which is a bad practice. Also this is a single place usage and I don't see need for this right now.

File: src/app/services/model-construct.service.ts (hunk 2 above)
Comment (svilenvelikov) [reply src/app/services/model-construct.service.ts:198]: repleced where applicable

**Pull request merged**